### PR TITLE
Improve sound handling for MMU error screen

### PR DESCRIPTION
**Pull request by gudnimg**
An idea I have to re-use M600 sound code for error screen.

The requirements are the same as for the M600 sound prompts.

* Loud mode => should continuously beep (loud prompt sound)
* Silent mode => should be silent
* Once mode => should beep only once
* Assist/Blind => Same as Loud mode but beep with slightly lower tone (same sound as when clicking or rotation the knob)

**TODO:**

- [x] Tune beeping period to be same as before in M600
- [x] Test M600 thoroughly (with all sounds settings)
- [x] Test MMU error screen (with all sound settings)

Testing:
* MMU Error screen:
    * Loud mode ✔️ 
    * Once mode ✔️  
    * Silent mode ✔️ 
    * Assist mode ✔️ 

Change in memory:
Flash: -148 bytes
SRAM: +4 bytes

PFW-831

### CODE REVIEW

File: Firmware/mmu2_reporting.cpp [hunk 1]
```diff
@@ -252,13 +253,7 @@ void ReportErrorHook(CommandInProgress /*cip*/, uint16_t ec, uint8_t /*es*/) {
...
                 break;
             case 2:
-                // Exit error screen and enable lcd updates
-                lcd_set_custom_characters();
-                lcd_update_enable(true);
-                lcd_return_to_status();
-                // Reset the state in case a new error is reported
-                mmu2.is_mmu_error_monitor_active = false;
-                ReportErrorHookState = ReportErrorHookStates::RENDER_ERROR_SCREEN;
+                ReportErrorHookState = ReportErrorHookStates::DISMISS_ERROR_SCREEN;
```
Comment (DRracer) [inline Firmware/mmu2_reporting.cpp]: How is this change related to sounds?

Comment (gudnimg) [reply Firmware/mmu2_reporting.cpp]: I would have needed to add a call to `sound_wait_for_user_reset();` here. When I wrote the error screen code this change above didn't save memory, but for some reason now it saves 10 bytes.

File: Firmware/mmu2_reporting.cpp [hunk 2]
```diff
@@ -256,6 +257,7 @@ void ReportErrorHook(CommandInProgress /*cip*/, uint16_t ec, uint8_t /*es*/) {
                 lcd_set_custom_characters();
```
Comment (DRracer) [inline Firmware/mmu2_reporting.cpp:257]: Are you going to extract this block into a separate function?
It is the same like `case (uint8_t)ReportErrorHookStates::DISMISS_ERROR_SCREEN:`

**Review by DRracer — APPROVED** [review]
